### PR TITLE
feat!: Change CLI to default to new SSA code for compilation

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

This PR does the minimal changes to move `nargo` over to use the new SSA code by default. You can continue to use the old SSA by using the `legacy-ssa` flag (which replaces `experimental-ssa`) although this should be removed soon.

## Documentation

- [x] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [x] I will request for and support Dev Rel's help in documenting this PR.
  
References to `experimental-ssa` should be removed. I would leave `legacy-ssa` undocumented as it will be removed soon.
## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Currently blocked by https://github.com/AztecProtocol/barretenberg/pull/626 as we want to get all ACIR tests passing in bberg before merging.

File: crates/nargo_cli/build.rs
Comment (jfecher) [inline crates/nargo_cli/build.rs:65]: Can we move the new tests into `test_data` and move the legacy tests into a new folder like `test_data_legacy` instead?

**Review by jfecher — COMMENTED** [review]
LGTM. I'll approve once the prerequisite PR is merged.

File: crates/nargo_cli/build.rs
Comment (TomAFrench) [reply crates/nargo_cli/build.rs:65]: I wanted to avoid moving around a lot of the various files as we can address this in future PRs as we remove the legacy SSA.

**Comment by TomAFrench** [comment]
On second thought we should probably push out a release before we merge this.

**Review by kevaundray — APPROVED** [review]
The blocking PR has been merged so approving

**Pull request merged**
